util/cloud/aws.cc

```diff
@@ -114,6 +122,196 @@ inline std::string_view std_sv(const ::boost::beast::string_view s) {
 
 constexpr char kAlgo[] = "AWS4-HMAC-SHA256";
 
+struct AwsConnectionData {
+  std::string access_key, secret_key, session_token;
+  std::string region;
+};
+
+// Try reading AwsConnectionData from env.
+std::optional<AwsConnectionData> GetConnectionDataFromEnv() {
+  const char* access_key = getenv("AWS_ACCESS_KEY_ID");
+  const char* secret_key = getenv("AWS_SECRET_ACCESS_KEY");
+  const char* session_token = getenv("AWS_SESSION_TOKEN");
+  const char* region = getenv("AWS_REGION");
+
+  if (access_key && secret_key) {
+    AwsConnectionData cd;
+    cd.access_key = access_key;
+    cd.secret_key = secret_key;
+    if (session_token)
+      cd.session_token = session_token;
+    if (region)
+      cd.region = region;
+    return cd;
+  }
+
+  return std::nullopt;
+}
+
+// Get path from ENV if env_var is set or default path relative to user home.
+std::optional<std::string> GetAlternativePath(std::string_view default_home_postfix,
+                                              const char* env_var) {
+  const char* path_override = getenv(env_var);
+  if (path_override)
+    return path_override;
+
+  const char* home_folder = getenv("HOME");
+  if (!home_folder)
+    return std::nullopt;
+
+  return absl::StrCat(home_folder, default_home_postfix);
+}
+
+std::optional<io::ini::Contents> ReadIniFile(std::string_view full_path) {
+  auto file = io::OpenRead(full_path, io::ReadonlyFile::Options{});
+  if (!file)
+    return std::nullopt;
+
+  io::FileSource file_source{file.value()};
+  auto contents = ::io::ini::Parse(&file_source, Ownership::DO_NOT_TAKE_OWNERSHIP);
+  if (!contents) {
+    LOG(ERROR) << "Failed to parse ini file:" << full_path;
+    return std::nullopt;
+  }
+
+  return contents.value();
+}
+
+// Try filling AwsConnectionData with data from config file.
+void GetConfigFromFile(AwsConnectionData& cd, const char* profile) {
```

Comment:
Google style-guide: output arguments should be last one and passed by a pointer. I.e here 'FillConfigFromFile(const char*, Aws...* cd)'